### ISSUE TITLE
Use Postgresql with Prisma for local dev, instead of SQLite

### ISSUE THREAD

**Issue by Martinsos** [wasp-lang/wasp#173]
Prisma pplz recommend that Postgresql is used both for local development and for production, since SQLite does not support all the features like Postgresql does, and you are immediately testing on what is also used in production which makes sense. SQLite is there really just for getting started quickly.

So, we should use Postgresql both in dev and prod -> right now in dev we use SQLite, and Postgre in prod.
Easiest way to use Postgre in dev is probably to run the docker container with postgre database, so nothing has to be installed on the machine, and Wasp can easily spin up this container during development and then turn it off when done.

Little bit of info on how to do this:
To run postgre database:
`docker run -p 5432:5432 --name psg-db -e POSTGRES_PASSWORD=test -d postgres` where we can specify db password (name is container name).
Default user is `postgres`, and it will store data somewhere on the disk on its own by default.
Default db name is also `postgres`.
Place where it stores data, user and db name can all be changed.
To connect to it, use `postgresql://<user>:<pass>@localhost:5432/<db-name>`.
So for command above, this would be `postgresql://postgres:test@localhost:5432/postgres`, and we can provide this as DATABASE_URL to prisma.

**Comment by Martinsos** [comment]
I should run the database with exact following command:
```
docker run --rm --publish 5432:5432 --env POSTGRES_PASSWORD=devpass postgres
```
this runs it in foreground, with logs being output to the CLI.
This would be a separate process/job.

**URL will be `postgresql://postgres:devpass@localhost:5432/postgres` -> we set this in schema.prisma.**

Question is, what about the data? Should I let it as it is -> will the database content persist? I guess not and I should use volumes. I should give it a try -> if it does not persist by default, I should go with specifying a volume. In that case, I guess I should add `-v wasp-db:/var/lib/postgresql/data`. `wasp-db` should probably be suffixed with the hash of the project or smth like that, to ensure that we have separation for different Wasp projects. Or, I could use just one volume, and have different names for the databases! Or, I could not use volumes but **use mount binds instead, and they would be bound to the specific location in the project directory -> maybe that is best for the start, since it is the easiest option**. We can play with volumes in the future if we will need better performance.

**Comment by Martinsos** [comment]
Ok, so let's try to nail exact steps down:

1. Extend DSL so that PSQL can be specified as a provider. TODO: figure out exact syntax.
2. When doing `wasp build`, if PSQL is not specified as a provider, abort and print error message.
3. When `schema.prisma` is being generated, check which provider is specified -> if none, use sqlite, if PSQL, use PSQL. For PSQL, use `postgresql://postgres:devpass@localhost:5432/postgres` as URL. Btw, here best way with setting schema.prisma to use PSQL might be to tell it to use env var and then providing that env var via .env.local.
4. When doing `wasp start` and PSQL is specified as a provider, run postgres docker as a job. Use bind mount to keep the data in the project dir. Use `docker run --rm --publish 5432:5432 --env POSTGRES_PASSWORD=devpass postgres` as a command.
5. When running db commands, make sure to also run postgres docker job in parallel, so that they can connect to the database.
6. If PSQL is specified as database and docker is not installed, produce nice error message.
7. Update docs.

**Comment by Martinsos** [comment]
Ok, after some reasoning about this, we decided to go with more flexible approach of having developer provide an URL to the database and that is it. That way they can use existing installation of postgres if they have it, or they can use docker command on their own, they can adjust database as they want, and so on. This means the whole experience is somewhat less managed, but we do give devs a lot of flexibility.
In the future, we might decide to do both: if you specify you want us to manage the db for you, we spin up the database for you via docker, if you don't, we let you provide it via DATABASE_URL.

So, in this case, we have following tasks:
- [ ] Extend DSL so that PSQL can be specified as provider. `db { provider: 'postgres' }`.
- [ ] When doing `wasp build`, if `postgres` is not specified as a provider, abort and print error message.
- [ ] When schema.prisma is being generated, check which provider is specified -> if none, use sqlite, if PSQL, use PSQL. For PSQL, read URL from env("DATABASE_URL").
- [ ] A whole separate task really: Add dot-env to the generated server code, and make sure that Wasp copies .env file to the generated code to server/. Also document this. And add .env to gitignore.
- [ ] Update docs -> create Database section. Give instructions on how to easily run a postgres database via single docker command (`docker run --rm --publish 5432:5432 --env POSTGRES_PASSWORD=devpass postgres`) + how to figure out the URL. TODO: Where do we bind data? Mention under deployment that database has to be postgres.

**Comment by Martinsos** [comment]
Prisma was upgraded with #217.

**Comment by Martinsos** [comment]
Done!
